### ISSUE TITLE
coq-doc-8.11.2-1: sphinx-build crashes while building html

### ISSUE THREAD

**Issue by cooljeanius** [fink/fink-distributions#1133]
Now that #880 has been fixed, I'm able to give upgrading coq-doc another try, but unfortunately that now runs into a new error:
```
SPHINXBUILD doc/sphinx (html)
Traceback (most recent call last):
  File "/opt/sw/lib/python3.8/site-packages/pkg_resources/__init__.py", line 573, in _build_master
    ws.require(__requires__)
  File "/opt/sw/lib/python3.8/site-packages/pkg_resources/__init__.py", line 891, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/opt/sw/lib/python3.8/site-packages/pkg_resources/__init__.py", line 782, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (charset-normalizer 3.3.2 (/opt/sw/lib/python3.8/site-packages), Requirement.parse('charset_normalizer<3,>=2'), {'requests'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/sw/bin/sphinx-build", line 33, in <module>
    sys.exit(load_entry_point('Sphinx==1.8.6', 'console_scripts', 'sphinx-build')())
  File "/opt/sw/bin/sphinx-build", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/opt/sw/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/opt/sw/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/sw/lib/python3.8/site-packages/sphinx/cmd/build.py", line 25, in <module>
    from sphinx.application import Sphinx
  File "/opt/sw/lib/python3.8/site-packages/sphinx/application.py", line 34, in <module>
    from sphinx.environment import BuildEnvironment
  File "/opt/sw/lib/python3.8/site-packages/sphinx/environment/__init__.py", line 38, in <module>
    from sphinx.util.websupport import is_commentable
  File "/opt/sw/lib/python3.8/site-packages/sphinx/util/websupport.py", line 11, in <module>
    from sphinxcontrib.websupport.utils import is_commentable  # NOQA
  File "/opt/sw/lib/python3.8/site-packages/sphinxcontrib/websupport/__init__.py", line 14, in <module>
    __import__('pkg_resources').declare_namespace(__name__)
  File "/opt/sw/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3267, in <module>
    def _initialize_master_working_set():
  File "/opt/sw/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3241, in _call_aside
    f(*args, **kwargs)
  File "/opt/sw/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3279, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/opt/sw/lib/python3.8/site-packages/pkg_resources/__init__.py", line 575, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/opt/sw/lib/python3.8/site-packages/pkg_resources/__init__.py", line 588, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/opt/sw/lib/python3.8/site-packages/pkg_resources/__init__.py", line 777, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'charset_normalizer<3,>=2' distribution was not found and is required by requests
make[3]: *** [Makefile.doc:72: refman-html] Error 1
make[3]: Leaving directory '/opt/sw/src/fink.build/coq-doc-8.11.2-1/coq-8.11.2'
make[2]: *** [Makefile:177: submake] Error 2
make[2]: Leaving directory '/opt/sw/src/fink.build/coq-doc-8.11.2-1/coq-8.11.2'
make[1]: *** [Makefile.doc:82: refman] Error 2
make[1]: Leaving directory '/opt/sw/src/fink.build/coq-doc-8.11.2-1/coq-8.11.2'
make: *** [Makefile:177: submake] Error 2
### execution of make failed, exit code 2
### execution of /tmp/fink.Yf_6i failed, exit code 2
Removing runtime build-lock...
Removing build-lock package...
/opt/sw/bin/dpkg-lockwait -r fink-buildlock-coq-doc-8.11.2-1
```
I'm on x86_64 Big Sur with Xcode 13.

**Comment by nieder** [comment]
Do you have charset-normalyzer-py38 installed?

**Comment by cooljeanius** [comment]
> Do you have charset-normalyzer-py38 installed?

Yes:
```
$ fink apropos charset-normal
Information about 15516 packages read in 2 seconds.
     charset-normalizer-py310                           3.3.2-1                                Real First Universal Charset Detector
 i   charset-normalizer-py37                            3.3.2-1                                Real First Universal Charset Detector
 i   charset-normalizer-py38                            3.3.2-1                                Real First Universal Charset Detector
     charset-normalizer-py39                            3.3.2-1                                Real First Universal Charset Detector
```

**Comment by nieder** [comment]
I updated requests-py, which was expecting an older charset-normalizer-py.
However, that leads to this error:
```
SPHINXBUILD doc/sphinx (html)
Running Sphinx v5.3.0

Warning, treated as error:
Invalid configuration value found: 'language = None'. Update your configuration to a valid language code. Falling back to 'en' (English).
```
I could patch that, but then the next error was about `cannot import name 'mathbase' from 'sphinx.ext'`, which is a deprecated and removed part of sphinx. There's a note in the .info file that newer versions of coq need need sphinx >= 2.3.1. We were stuck at an older sphinx for a long time, but we now have v5.3.0. It looks like our old coq-doc is unhappy with the newer docbuild tools. Will see about updating it.

**Comment by nieder** [comment]
See #1134 where I updated coq to the latest upstream.

**Comment by cooljeanius** [comment]
> See #1134 where I updated coq to the latest upstream.

OK yeah, that's probably good to merge.

**Comment by nieder** [comment]
Closed by #1134
